src/arch/io/disk.cc

```diff
@@ -482,7 +482,11 @@ file_open_result_t open_file(const char *path, const int mode, io_backender_t *b
 #ifdef __linux__
     flags |= O_LARGEFILE;
 #endif
-
+#if defined(__FreeBSD__)
+    // FreeBSD does not have O_DSYNC / F_FULLSYNC. Instead, it relies on file
+    // system features like "soft updates" to reduce metadata syncing time.
+    flags |= O_SYNC;
```

Comment:
Do you know if we actually need the O_SYNC here? Maybe just the `fsync` call in  line 632 is enough?
If in doubt, we can leave the O_SYNC in. It might be necessary for safety, and in the worst case will cost performance.